programs/drift/src/math/margin.rs

```diff
@@ -525,13 +562,7 @@ pub fn meets_withdraw_margin_requirement(
         )?;
     }
 
-    if calculation.get_num_of_liabilities()? > 1 {
-        validate!(
-            !calculation.with_isolated_liability,
-            ErrorCode::IsolatedAssetTierViolation,
-            "User attempting to increase number of liabilities above 1 with a isolated tier liability"
-        )?;
-    }
+    validate_any_isolated_tier_requirements(user, calculation)?;
```

Comment:
think we need to add this to fill_spot_order?

Reply:
you also need to block that user from opening new usdc borrow, i think this code would let it thru

Reply:
also user could have a big usdc borrow and this would still let them open perp position

Reply:
resolved given isolated allows usdc borrow